### ISSUE TITLE
Easy proxy config with defaults or inline

### ISSUE THREAD

**Issue by xealot** [angular/angular-cli#6240]
### Bug Report or Feature Request
- [ ] bug report -> please search issues before submitting
- [x] feature request

### Versions.
```
@angular/cli: 1.0.2
node: 7.9.0
os: darwin x64
@angular/animations: 4.1.0
@angular/common: 4.1.0
@angular/compiler: 4.1.0
@angular/core: 4.1.0
@angular/forms: 4.1.0
@angular/http: 4.1.0
@angular/platform-browser: 4.1.0
@angular/platform-browser-dynamic: 4.1.0
@angular/router: 4.1.0
@angular/cli: 1.0.2
@angular/compiler-cli: 4.1.0
```

### Desired functionality.
**Level 1**
Allow us to specify `--proxy-conf` command line flag through `.angular-cli.json` defaults mechanism.

**Level 2**
Allow inlining of proxy config into `.angular-cli.json` to reduce the amount of files in the root.

**Comment by filipesilva** [comment]
I think the first bit is a useful setting for projects that always need to be ran with that proxy. I'd prefer not to inline it though, since it can also be a `.js` file and it could massively complicate the CLI json file schema.

**Comment by ayvazj** [comment]
I submitted a PR for this HTH https://github.com/angular/angular-cli/pull/6493

**Comment by shairez** [comment]
@hansl @Brocco This can be closed (it was already merged by #6493 )
